chapters/en/slides/module3/module3_02.Rmd

```diff
@@ -0,0 +1,308 @@
+---
+params:
+  dynamictitle: "module3_02"
+title: "`r params$dynamictitle`"
+output:
+  md_document:
+    variant: gfm
+    pandoc_args: "--markdown-headings=atx"
+---
+
+```{r setup, include=FALSE}
+source('../../../../slide-setup.R')
+# Remember to also manually update the YAML title above
+```
+
+type: slides
+
+# Using density plots to visualize distributions
+
+Notes:
+In this module we will see how to use density plots
+instead of histograms to represent distributions of data.
+
+---
+
+## Histograms can give different results depending on the data
+
+<img src=/module3/histogram-and-shifted.png></img>
+
+[From scikit-learn](https://scikit-learn.org/stable/auto_examples/neighbors/plot_kde_1d.html)
+
+Notes:
+How many observations are counted per bar in a histogram
+depends on exactly where on the axis the border between the bins are.
+
+In the images on this slide,
+the actual observations are drawn with black tick marks on the bottom
+and they are the same in both subplots.
+
+The reason the histograms look different
+is that the border between the groups is shifted in the rightmost picture.
+
+As we see here,
+a histogram is not an as unbiased plot as we might think at first,
+especially not if we have few data points,
+where the inclusion or exclusion of just a few points
+makes a big difference for the bar height.
+
+---
+
+## Centering the bins on the data can help create more accurate distribution plots
+
+<img src=/module3/kde-buildup.gif></img>
+
+Notes:
+Instead of setting fixed lines along the axis
+and then count plots fully in one bin or another,
+we can create bins that are centered
+on the data and then add them together.
+
+When we center bins on the data,
+we also usually have a round shape 
+instead of a square one,
+which gives the impression of a smoother distribution
+that is more conducive to communicating the shape of the data distribution.
+
+The round bins (also called kernels)
+are then added together as in the gif in this slide,
+and sum up to an overall distribution line.
+Formally,
+this is called a Kernel Density Estimate (KDE),
+or just a density plot.
+
+---
+
+## The movies dataset
+
+```{python}
+import pandas as pd
+
+
+movies = pd.read_csv('movies.csv')
+movies
+```
+
+Notes:
+We will be using the movies dataset again,
+let's see if we can make a plot
+that is easier to interpret
+than the layered histograms
+from the end of the last slide deck.
+
+---
+
+## Creating a density plot in Altair required two step
+
+```{python}
+import altair as alt
+
+(alt.Chart(movies).transform_density(
+    'runtime',
+    as_=['runtime', 'density'])  # Give a name to the KDE values, which we can use when plotting
+ .mark_area().encode(
+    x='runtime',
+    y='density:Q'))
+```
+
+Notes:
+To visualize a density plot,
+we need to do two things:
+
+1. Calculate the KDE (adding the individual kernels and summing them together)
```

Comment:
Don't use KDE with these learners, too much new stuff is going on already, the do not need an acronym to remember as well.

Reply:
Good catch, I will switch to something like "adding the rounded bins together"